src/box/lua/net_box.lua

```diff
@@ -303,14 +306,36 @@ local function new_sm(uri_or_fd, opts)
                 remote._shutdown_pending = nil
             end
             if state == 'error_reconnect' then
-                -- Repeat the same error in verbose log only.
-                -- Else the error clogs the log. See gh-3175.
-                if err ~= last_reconnect_error then
-                    log.warn('%s:%s: %s', host or "", port or "", err)
-                    last_reconnect_error = err
+                local function display_warning()
```

Comment:
Nit: please, don't inline functions. It's better to give it a proper name and place it somewhere above